client/extensions/woocommerce/state/sites/settings/email/test/reducer.js

```diff
@@ -0,0 +1,186 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import {
+	WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST,
+	WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST_SUCCESS,
+	WOOCOMMERCE_MAILCHIMP_API_KEY_SUBMIT_FAILURE,
+	WOOCOMMERCE_MAILCHIMP_API_KEY_SUBMIT_SUCCESS,
+	WOOCOMMERCE_MAILCHIMP_API_KEY_SUBMIT,
+	WOOCOMMERCE_MAILCHIMP_LISTS_REQUEST_SUCCESS,
+	WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST_FAILURE,
+} from 'woocommerce/state/action-types';
+import reducer from 'woocommerce/state/sites/reducer';
+
+describe( 'reducer', () => {
+	test( 'should mark the settings request fetching', () => {
+		const siteId = 123;
+		const action = {
+			type: WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST,
+			siteId,
+		};
+
+		const newSiteData = reducer( {}, action );
+		expect( newSiteData[ siteId ].settings.email.settingsRequest ).to.eql( true );
+	} );
+
+	test( 'should store data from the action', () => {
+		const siteId = 123;
+		const settings = { store_name: 'mystore' };
+		const action = {
+			type: WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST_SUCCESS,
+			siteId,
+			settings,
+		};
+		const newState = reducer( {}, action );
+		expect( newState[ siteId ] ).to.exist;
+		expect( newState[ siteId ].settings ).to.exist;
+		expect( newState[ siteId ].settings.email.settings ).to.deep.equal( settings );
+	} );
+
+	test( 'should merge lists data into settigns and pick first as default if none is chosen', () => {
+		const siteId = 123;
+		const settings = { store_name: 'mystore' };
+		const lists = { a: 'list_a', b: 'list_b' };
+		const settingsAction = {
+			type: WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST_SUCCESS,
+			siteId,
+			settings,
+		};
+
+		const listsAction = {
+			type: WOOCOMMERCE_MAILCHIMP_LISTS_REQUEST_SUCCESS,
+			siteId,
+			lists,
+		};
+
+		const finalState = {
+			store_name: 'mystore',
+			mailchimp_lists: {
+				a: 'list_a',
+				b: 'list_b',
+			},
+			mailchimp_list: 'a',
+		};
+
+		const firstState = reducer( {}, settingsAction );
+		const updatedState = reducer( firstState, listsAction );
+		expect( updatedState[ siteId ] ).to.exist;
+		expect( updatedState[ siteId ].settings ).to.exist;
+		expect( updatedState[ siteId ].settings.email.settings ).to.deep.equal( finalState );
+	} );
+
+	test( 'should merge lists data into settigns and not update chosen list', () => {
+		const siteId = 123;
+		const settings = { store_name: 'mystore', mailchimp_list: 'c' };
+		const lists = { a: 'list_a', b: 'list_b' };
+		const settingsAction = {
+			type: WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST_SUCCESS,
+			siteId,
+			settings,
+		};
+
+		const listsAction = {
+			type: WOOCOMMERCE_MAILCHIMP_LISTS_REQUEST_SUCCESS,
+			siteId,
+			lists,
+		};
+
+		const finalState = {
+			store_name: 'mystore',
+			mailchimp_lists: {
+				a: 'list_a',
+				b: 'list_b',
+			},
+			mailchimp_list: 'c',
```

Comment:
Remind me what the `mailchimp_list` state represents? In the real world, would it not be either 'a' or 'b' here?

Reply:
You have a very good eye :) This is a situation that is considered an error inside the MailChimp plugin - it would manifest itself in different places. I have used it to properly tests the plugin and make sure that the `c` is not overwritten. In normal operation it would be `a` or `b` 
`mailchimp_list` is the list that is chosen by plugins user to be his chosen list.
`mailchimp_lists` ( plural ) it the list of mailing lists configured for the account at mailchimp.com